src/librepgp/stream-parse.cpp

```diff
@@ -859,51 +859,40 @@ signed_read_single_signature(pgp_source_signed_param_t *param,
                              pgp_source_t *             readsrc,
                              pgp_signature_t **         sig)
 {
-    uint8_t               ptag;
-    int                   ptype;
-    pgp_signature_t       readsig;
-    pgp_signature_info_t *siginfo;
-
+    uint8_t ptag;
     if (!src_peek_eq(readsrc, &ptag, 1)) {
         RNP_LOG("failed to read signature packet header");
         return RNP_ERROR_READ;
     }
 
-    ptype = get_packet_type(ptag);
-
+    int ptype = get_packet_type(ptag);
     if (ptype != PGP_PKT_SIGNATURE) {
         RNP_LOG("unexpected packet %d", ptype);
         return RNP_ERROR_BAD_FORMAT;
     }
 
     try {
         param->siginfos.push_back({});
```

Comment:
don't we use `emplace_back()` in such cases?

Reply:
Actually it's leftover from some other PR, but let's fix it as well.